src/serac/numerics/functional/tuple_arithmetic.hpp

```diff
@@ -151,23 +151,146 @@ SERAC_HOST_DEVICE auto chain_rule_tuple_matvec(serac::tuple<T...> df_dx, serac::
 }
 
 }  // namespace detail
+
+template <int i, typename S, typename T>
+struct one_hot_helper;
```

Comment:
What is `hot` in this context?

Reply:
it's a reference to [one-hot encoding](https://en.wikipedia.org/wiki/One-hot), where you have a list of entries, all of which are "off" except for one

Reply:
k. Can you add this to the doxygen comments please?

Reply:
this is explained in more detail on `one_hot_t` (which is the type that is actually intended to be used):
```
/**
 * @brief a tuple type with n entries, all of which are of type `serac::zero`,
 * except for the i^{th} entry, which is of type T
 *
 *  e.g. one_hot_t< 2, 4, T > == tuple<zero, zero, T, zero>
 */
template <int i, int n, typename T>
using one_hot_t = typename one_hot<i, n, T>::type;
```
Unfortunately, these other types (e.g. `one_hot`, `one_hot_helper`) are boilerplate that are practically required by the language to implement this sort of thing (`one_hot_t`). 

Up until now, I've just been documenting the main type (e.g. `one_hot_t`) to avoid redundant doxygen comments, but I could start including those additional comments if you think it's helpful.

Reply:
While it involves duplication, I think its useful to have the doxygen here as well so developers don't have to go digging for the explanation.

Reply:
Or just say go look at this other struct